modules/031-linstor/images/linstor-pools-importer/main.go

```diff
@@ -0,0 +1,559 @@
+)
+
+const (
+	lvmConfig      = `devices {filter=["r|^/dev/drbd*|"]}`
+	linstorPrefix  = "linstor"
+	maxReplicasNum = 3
+)
+
+// Print version
+func printVersion() {
+	klog.Info(fmt.Sprintf("Go Version: %s", runtime.Version()))
+	klog.Info(fmt.Sprintf("Go OS/Arch: %s/%s", runtime.GOOS, runtime.GOARCH))
+}
+
+type options struct {
+	nodeName     string
+	scanInterval int
+}
+
+func main() {
+
+	// Parse inputs
+	opts, err := parseInputs()
+	if err != nil {
+		klog.Fatalln("Failed to parse inputs", err)
+	}
+
+	// Print version
+	printVersion()
+
+	// Create Kubernetes client
+	kc, err := createKubeClient()
+	if err != nil {
+		klog.Fatalln("...", err)
+	}
+
+	// Create context
+	ctx, cancel := context.WithCancel(context.Background())
+
+	// Get owner
+	owner, err := getOwner(ctx, kc, opts.nodeName)
+	if err != nil {
+		klog.Fatalln("failed to get owner pod:", err)
+	}
+
+	// Get LINSTOR client
+	lc, err := lclient.NewClient()
+	if err != nil {
+		klog.Fatalln("failed to create LINSTOR client:", err)
+	}
+
+	klog.Infof("Starting main loop (scanInterval: %d seconds)", opts.scanInterval)
+
+	// Main loop: configure storage pools and create storage classes
+	stop := make(chan struct{})
+	go func() {
+		defer cancel()
+		err := provisionStoragePools(ctx, lc, kc, &owner, opts.nodeName, opts.scanInterval)
+		if errors.Is(err, context.Canceled) {
+			// only occurs if the context was cancelled, and it only can be cancelled on SIGINT
+			stop <- struct{}{}
+			return
+		}
+		klog.Fatalln(err)
+	}()
+
+	// Blocks waiting signals from OS.
+	ch := make(chan os.Signal, 1)
+	signal.Notify(ch, syscall.SIGINT, syscall.SIGTERM)
+
+	<-ch
+	cancel()
+	<-stop
+}
+
+// Parse inputs
+func parseInputs() (options, error) {
+	var opts options
+	opts.scanInterval = *flag.Int("scan-interval", 10, "Delay in seconds between scanning attempts")
+	klog.Infof("scanInterval: %d", opts.scanInterval)
+
+	opts.nodeName = os.Getenv("NODE_NAME")
+	if opts.nodeName == "" {
+		return opts, fmt.Errorf("Required NODE_NAME env variable is not specified!")
+	}
+	klog.Infof("nodeName: %s", opts.nodeName)
+
+	flag.Parse()
+	return opts, nil
+}
+
+// Create Kubernetes client
+func createKubeClient() (kclient.Client, error) {
+	var kc kclient.Client
+	clientConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+		clientcmd.NewDefaultClientConfigLoadingRules(),
+		&clientcmd.ConfigOverrides{},
+	)
+	// Get a config to talk to the apiserver
+	config, err := clientConfig.ClientConfig()
+	if err != nil {
+		return kc, fmt.Errorf("Failed to get kubernetes config: %w", err)
+	}
+	kc, err = kclient.New(config, kclient.Options{})
+	if err != nil {
+		return kc, fmt.Errorf("Failed to create Kubernetes client: %w", err)
+	}
+	return kc, nil
+}
+
+// Get owner
+func getOwner(ctx context.Context, kc kclient.Client, nodeName string) (v1.ObjectReference, error) {
+	var owner v1.ObjectReference
+	var node v1.Node
+	err := kc.Get(ctx, types.NamespacedName{Name: nodeName}, &node)
```

Comment:
Do you need a UID for a node here? Is it required? If not, I would like to propose going with a node name alone.

Reply:
OK, it seems that involved object does nothing except improving readability of events, let's use storagepools and storageclasses as involvedObject with no uid.

Reply:
```
2m22s       Normal    Created               storagepool/hf-kubevirt-01.hdd   Created LINSTOR storage pool: hf-kubevirt-01/hdd
2m20s       Normal    Created               storagepool/hf-kubevirt-02.hdd   Created LINSTOR storage pool: hf-kubevirt-02/hdd
2m19s       Normal    Created               storagepool/hf-kubevirt-03.hdd   Created LINSTOR storage pool: hf-kubevirt-03/hdd
19s         Normal    Created               storageclass/linstor-hdd-r1      Created Kubernetes storage class: linstor-hdd-r1
19s         Normal    Created               storageclass/linstor-hdd-r2      Created Kubernetes storage class: linstor-hdd-r2
19s         Normal    Created               storageclass/linstor-hdd-r3      Created Kubernetes storage class: linstor-hdd-r3
```